### PR TITLE
Added K-means algorithm using C

### LINKED ISSUE #1795: Implementation of K-means clustering algorithm using C
Would like to add the K-means clustering algorithm implemented in C.
@asha15 @singh-shreya6 @jainaman224 @raghav-dalmia please assign me this

### PR DESCRIPTION
**Pull request by arghac14**
Fixes #1795 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added the KNN Algorithm implemented using C to the Machine_Learning directory.

#### Languages Used:
-  C

#### Files Added:
-  K-means/K-means_using_c.c

> We're happy to help you get this ready -- don't be afraid to ask for help.

@SakshiShreya @singh-shreya6 @asha15 @raghav-dalmia @jainaman224 @MastersAbh please review the PR.
Thanks!

### CODE REVIEW

File: Machine_Learning/K-means/k-means_using_c.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	//calculating new mean
+    for(t1=0;t1<j;t1++){
+        t2=t2+a1[t1];
+    }
+    m1=t2/k;
+    t2=0;
+    for(t1=0;t1<k;t1++){
+        t2=t2+a2[t1];
+    }
+    m2=t2/k;
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep space around operator

File: Machine_Learning/K-means/k-means_using_c.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        printf("%d ",a1[t1]);
+    }
+    printf("\nm1= %d",m1);
+    printf("\nCluster 2: ");
+    for(t1=0;t1<k;t1++)
+    {
+        printf("%d ",a2[t1]);
+    }
+    printf("\nm2= %d",m2);
+    }while(m1!=om1&&m2!=om2);
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep space around operator

File: Machine_Learning/K-means/k-means_using_c.c [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+//Enter second mean: 7
+//
+//Cluster 1: 1 3 2
+//m1= 0
+//Cluster 2: 6 8 5 7 8 5 9
+//m2= 6
+//Cluster 1: 1 2
+//m1= 0
+//Cluster 2: 6 8 3 5 7 8 5 9
+//m2= 6
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: make this as a multi line commend

**Comment by arghac14** [comment]
@asha15 I have made the required changes, please review.

**Comment by arghac14** [comment]
@asha15 I have made the required changes, please review

File: Machine_Learning/K-means/k-means_using_c.c [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        if(t1 < t2){
+            //near to first mean
+            a1[j] = a0[i];
+            j++;
+        }
+        else
+        {
+            //near to second mean
+            a2[k] = a0[i];
+            k++;
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: correct the indentation

**Comment by arghac14** [comment]
@asha15 ma'am I Will correct the indentation about the extra file, every time I'm squashing and rebasing the commits that empty git file comes out of nowhere. i think it's a bug of my OS or the cmder tool I'm using.  So should I make a new PR or multiple commits will work without squashing? Cause I can delete the file. Please let me know!

File: Machine_Learning/K-means/k-means_using_c.c [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        }
+    }
+    t2 = 0;
+    
+    //calculating new mean
+    for(t1 = 0; t1 < j; t1++){
+        t2 = t2 + a1[t1];
+    }
+    m1 = t2 / k;
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space before declaring

File: Machine_Learning/K-means/k-means_using_c.c [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    t2 = 0;
+    
+    for(t1 = 0; t1 < k; t1++){
+        t2 = t2 + a2[t1];
+    }
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    m2 = t2 / k;
+    
+    //printing clusters
+    printf("\nCluster 1: ");
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    for(t1 = 0; t1 < j; t1++){
+        printf("%d ", a1[t1]);
+    }
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    printf("\nm1= %d", m1);
+
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: remove this line

Comment (asha15) [reply Machine_Learning/K-means/k-means_using_c.c]: keep space around equal

File: Machine_Learning/K-means/k-means_using_c.c [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    m2 = t2 / k;
+    
+    //printing clusters
+    printf("\nCluster 1: ");
+    for(t1 = 0; t1 < j; t1++){
+        printf("%d ", a1[t1]);
+    }
+    printf("\nm1= %d", m1);
+    
+    printf("\nCluster 2: ");
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    for(t1 = 0; t1 < k; t1++){
+        printf("%d ", a2[t1]);
+    }
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    scanf("%d", &m2);
+     
+    int om1, om2;    //previous means
+    
+    do{
+    om1 = m1;
+    om2 = m2;
+    
+    //creating clusters
+    i = j = k = 0;
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+    for(i = 0; i < 10; i++){
+        //calculating distance to means
+        t1 = a0[i] - m1;
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+        if(t1 < 0){
+            t1 = -t1;
+        }
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+    
+    //creating clusters
+    i = j = k = 0;
+    for(i = 0; i < 10; i++){
+        //calculating distance to means
+        t1 = a0[i] - m1;
+        if(t1 < 0){
+            t1 = -t1;
+        } 
+        t2 = a0[i] - m2;
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

File: Machine_Learning/K-means/k-means_using_c.c [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+        if(t2 < 0){
+            t2 = -t2;
+        }
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: keep a line space after this

**Comment by asha15** [comment]
just try `git add Machine_Learning/K-means/k-means_using_c.c` command instead if `git add .`
I hope it will solve the problem

please correct this minor changes and let me know

**Comment by arghac14** [comment]
@asha15 ma'am `git add Machine_Learning/K-means/k-means_using_c.c` didn't solve the extra empty file issue, I have made the changes so please let me know is it okay with 2 commits or should I try to make a new PR and see if that works.

git issue

**Comment by arghac14** [comment]
@asha15 ma'am I have created a new PR with all the changes, please see and let me know if any changes required.
https://github.com/jainaman224/Algo_Ds_Notes/pull/2417

**Pull request closed without merging**